scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -1429,7 +1429,7 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
   def inlineMatchClause(inlineMods: List[Mod]) = {
     autoEndPos(inlineMods)(postfixExpr(allowRepeated = false)) match {
       case t: Term.Match =>
-        copyPos(t)(t.copy(mods = inlineMods))
+        copyPos(t)(t.fullCopy(mods = inlineMods))
```

Comment:
Why do we need fullCopy here?

Reply:
for now, `fullCopy` does exactly the same as the primary `copy`. However, after we introduce a new annotation, for replaced fields, the primary `.copy` in those cases will be producing trees with replaced fields losing their origin. which might be ok for outside usage but not good for this parser.